src/sources/mongodb_metrics/mod.rs

```diff
@@ -273,13 +196,23 @@ impl MongoDBMetrics {
         })
     }
 
-    async fn get_build_info(client: &Client) -> Result<CommandBuildInfo, BuildError> {
-        let doc = client
+    async fn get_build_info(&self) -> Result<CommandBuildInfo, CollectError> {
+        let doc = self
+            .client
             .database("admin")
             .run_command(doc! { "buildInfo": 1 }, None)
             .await
-            .context(CommandBuildInfoMongoError)?;
-        from_document(doc).context(CommandBuildInfoParseError)
+            .map_err(CollectError::Mongo)?;
+        from_document(doc).map_err(CollectError::Bson)
+    }
+
+    async fn print_version(&self) -> Result<(), CollectError> {
+        let node_type = self.get_node_type().await?;
+        let build_info = self.get_build_info().await?;
+        debug!(
```

Comment:
Given this message is only printed at the debug level, would it make sense to not even fetch the node type and build info if debug logs are not enabled?

I'm also fine with dropping this log message in-lieu of debug messages during the actual collection.

Reply:
Updated for mongo and postgres :slightly_smiling_face: